### ISSUE TITLE
Rule proposal: prefer `Symbol.for()` over `Symbol()`

### ISSUE THREAD

**Issue by Xunnamius** [sindresorhus/eslint-plugin-unicorn#2027]
### Description

From time to time I find myself passing things between realms, usually in Node/Jest but also between [workers](https://html.spec.whatwg.org/multipage/workers.html#worker-processing-model) or frames on the frontend, and I've been bitten by [faulty `instanceof` checks (identity discontinuity)](https://github.com/sindresorhus/eslint-plugin-unicorn/issues/723#issuecomment-626791865) more than once. My solution has been to use Symbols to track an object's type/provenance.

For instance, for a piece of code meant to handle a subset of custom app errors, instead of `catch(e) { if(e instanceof AppError) ... }` I'll do `catch(e) { if(e?.[$type] === 'AppError') ... }` where `const $type = Symbol('type')` and `class AppError extends Error { [$type] = 'AppError'; ... }`.

However, unlike the well-known Symbols, custom Symbols are only unique at the [realm level](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol#shared_symbols_in_the_global_symbol_registry) and so suffer the same identity discontinuity issue as `instanceof`. But we can create cross-realm Symbols with [`Symbol.for`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/for). Perhaps `Symbol.for()` should be preferred over `Symbol()` in all cases?

### Fail

```js
const foo = Symbol('foo');
```

### Pass

```js
const foo = Symbol.for('foo');
```

```js
const foo = Symbol.for('@Xunnamius/foo');
```

### Additional Info

N/A

**Comment by fisker** [comment]
They are for different purposes. 

As for the  `instanceof` problem, you man want try [`Ergonomic brand checks for Private Fields`](https://github.com/tc39/proposal-private-fields-in-in)

**Comment by Xunnamius** [comment]
I see. Thanks for the link

**Comment by sindresorhus** [comment]
Passing on this as it's not the correct use of `Symbol.for`.
